fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -1068,8 +1068,13 @@ private void analyzeAggregation(Analyzer analyzer) throws AnalysisException {
 
             substituteOrdinalsAliases(groupingExprs, "GROUP BY", analyzer);
 
-            if (!groupByClause.isGroupByExtension()) {
+            if (!groupByClause.isGroupByExtension() && !groupingExprs.isEmpty()) {
+                ArrayList<Expr> tempExprs = new ArrayList<>(groupingExprs);
                 groupingExprs.removeIf(Expr::isConstant);
+                if (groupingExprs.isEmpty()) {
+                    // should keep at least one expr to make the result correct
+                    groupingExprs.add(tempExprs.get(0));
```

Comment:
why we need to keep one? i think 
```
select sum(a) from t group by 2
```
is same with 
```
select sum(a) from t
```

Reply:
yes, originally, this pr is to correct "select 2 from nation group by 1". But as you mentioned, if there is any agg exprs, the grouping exprs can be removed safely, I updated the code to reflect that. thanks.